examples/statemachine/src/language-server/statemachine-validator.ts

```diff
@@ -29,20 +38,24 @@ export class StatemachineValidator {
         if (state.name) {
             const firstChar = state.name.substring(0, 1);
             if (firstChar.toUpperCase() !== firstChar) {
-                accept('warning', 'State name should start with a capital letter.', { node: state, property: 'name' });
+                accept('warning', 'State name should start with a capital letter.', {
+                    node: state,
+                    property: 'name',
+                    data: diagnosticData(IssueCodes.StateNameUppercase),
+                });
             }
         }
     }
 
     /**
-     * Checks if there are duplicate state and event names.
+     * Checks if there are duplicate command, event, and state names.
      * @param statemachine the statemachine to check
      * @param accept the acceptor to report errors
      */
-    checkUniqueStatesAndEvents(statemachine: Statemachine, accept: ValidationAcceptor): void {
+    checkUniqueSymbolName(statemachine: Statemachine, accept: ValidationAcceptor): void {
```

Comment:
I'd suggest to have separate name pools for each entity: `Command`, `Event` and `State`. It can resolve such situations like
* This is an error
<img width="164" alt="Screenshot 2023-11-09 at 16 27 48" src="https://github.com/eclipse-langium/langium/assets/15619772/2ff79970-ba34-40e4-9927-7efd1519592d">

* But this is correct
<img width="167" alt="Screenshot 2023-11-09 at 16 31 30" src="https://github.com/eclipse-langium/langium/assets/15619772/88576dc9-0ad0-472c-a441-8cdd194efd5a">